Correct URL encoding of CSS

annevk opened the pull request:
#### c195704a1cc8afc565fb5bd694d06fd329321267
<pre>
Correct URL encoding of CSS
<a href="https://bugs.webkit.org/show_bug.cgi?id=261079">https://bugs.webkit.org/show_bug.cgi?id=261079</a>
<a href="https://rdar.apple.com/114889625">rdar://114889625</a>

Reviewed by Tim Nguyen.

CSS invokes the URL parser normally and therefore always uses UTF-8 as
its URL encoding: <a href="https://drafts.csswg.org/css-values/#url-processing.">https://drafts.csswg.org/css-values/#url-processing.</a>

This aligns us with the standard and Gecko.

* LayoutTests/http/tests/webarchive/test-css-url-encoding-expected.webarchive:
* LayoutTests/http/tests/webarchive/test-css-url-encoding-shift-jis-expected.webarchive:
* LayoutTests/http/tests/webarchive/test-css-url-encoding-utf-8-expected.webarchive:
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/utf-8_include=css-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/windows-1251_include=css-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/windows-1252_include=css-expected.txt:
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::completeURL const):

Canonical link: <a href="https://commits.webkit.org/270169@main">https://commits.webkit.org/270169@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8f5bd704893fa52f77884e474ecfd147cc1319c1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24745 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3290 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25998 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26863 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22717 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/25014 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4965 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/727 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23069 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24990 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2339 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21363 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27446 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/2046 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22294 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28454 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22569 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22636 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26264 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1981 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/291 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3277 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5926 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2433 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2339 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
r=me with relevant tests rebaselined since the spec change has been approved

annevk committed:
Correct URL encoding of CSS
https://bugs.webkit.org/show_bug.cgi?id=261079
rdar://114889625

Reviewed by Tim Nguyen.

CSS invokes the URL parser normally and therefore always uses UTF-8 as
its URL encoding: https://drafts.csswg.org/css-values/#url-processing.

This aligns us with the standard and Gecko.

* LayoutTests/http/tests/webarchive/test-css-url-encoding-expected.webarchive:
* LayoutTests/http/tests/webarchive/test-css-url-encoding-shift-jis-expected.webarchive:
* LayoutTests/http/tests/webarchive/test-css-url-encoding-utf-8-expected.webarchive:
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/utf-8_include=css-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/windows-1251_include=css-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/windows-1252_include=css-expected.txt:
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::CSSParserContext::completeURL const):

Canonical link: https://commits.webkit.org/270169@main
